doc/ug/rust_for_c.md

```diff
@@ -0,0 +1,2310 @@
+However, Unsafe Rust has Undefined Behavior, and, due to its stronger invariants, Rust is generally more punishing than C compilers when Undefined Behavior occurs.
+
+[^3]: Alternative implementations exist, though `rustc` is the reference implementation.
+Rust is still in the process of being specified, so, for now, `rustc`'s behavior is mostly normative.
+
+[^4]: Not every project uses rustup, but it's frequently necessary for "nightly" features.
+
+[^5]: Inline assembly is the most salient of these.
+Nightly also provides support for the sanitizers, such as ASAN and TSAN.
+
+[^6]: Example from Tock: [https://github.com/tock/tock/blob/master/rust-toolchain](https://github.com/tock/tock/blob/master/rust-toolchain)
+
+[^7]: Rust libraries use their own special "rlib" format, which carries extra metadata past what a normal .a file would.
+
+[^8]: While each crate is essentially a giant object file, Rust will split up large crates into smaller translation units to aid compilation time.
+
+[^9]: The formatter is kind of a new component, so may require separate installation through `rustup`.
+See [https://github.com/rust-lang/rustfmt#on-the-stable-toolchain](https://github.com/rust-lang/rustfmt#on-the-stable-toolchain).
+
+[^10]: [https://doc.rust-lang.org/core/index.html](https://doc.rust-lang.org/core/index.html)
+
+[^11]: They are also used analogously to `ptrdiff_t` and `size_t`.
+In practice, these types are all the same width on modern systems, though C draws a distinction for portability reasons.
+
+[^12]: Note that Rust spells bitwise negation on integers as `!x` rather than `~x`.
+This is because Rust has a separate boolean type, for which `!x` is logical not.
+
+[^13]: Rust has slightly less absurd precedence rules around bitwise operators; `x ^ y == z` is `(x ^ y) == z`.
+
+[^14]: In C, signed overflow is UB, and unsigned overflow always wraps around.
+
+[^15]: Really, this causes a "panic", which triggers unwinding of the stack.
+In embedded contexts, unwinding is disabled, so this is just a jump to the panic handler, which performs a platform-specific abort.
+
+[^16]: rustc performs the former in debug builds and the latter in release builds.
+
+[^17]: Creating `bool`s with any other representation, using Unsafe Rust, is instant UB.
+This is sometimes called the "validity invariant", and applies to other restricted-range types, such as padding bytes in structs.
+["What The Hardware Does" is not What Your Program Does: Uninitialized Memory](https://www.ralfj.de/blog/2019/07/14/uninit.html) provides a short explanation about why this is important.
+
+[^18]: These functions are frequently entry-points for LLVM intrinsics, similar to GCC/Clangs `__builtin_clz()` and similar.
+
+[^19]: This syntax: `(my_struct_t) { .a = 0, .b = 1 }`.
+
+[^20]: Rust allows trailing commas basically everywhere.
```

Comment:
Something is screwed up around footnote 19/20. Either could successfully be applied where [^19] is in the text, but afterwards all the remaining footnotes are off by one. For example [^20] in the text should really be [^21].

Reply:
Still pretty sure your footnotes are off-by-one starting at [^20]. I don't think this is just a github visualization issue.